iina/MPVController.swift

```diff
@@ -153,6 +153,66 @@ class MPVController: NSObject {
     removeOptionObservers()
   }
 
+
+  /// Determine if this Mac has an Apple Silicon chip.
+  /// - Returns: `true` if running on a Mac with an Apple Silicon chip, `false` otherwise.
+  private func runningOnAppleSilicon() -> Bool {
+    // Old versions of macOS do not support Apple Silicon.
+    if #unavailable(macOS 11.0) {
+      return false
+    }
+    var sysinfo = utsname()
+    let result = uname(&sysinfo)
+    guard result == EXIT_SUCCESS else {
+      Logger.log("uname failed returning \(result)", level: .error)
+      return false
+    }
+    let data = Data(bytes: &sysinfo.machine, count: Int(_SYS_NAMELEN))
+    guard let machine = String(bytes: data, encoding: .ascii) else {
+      Logger.log("Failed to construct string for sysinfo.machine", level: .error)
+      return false
+    }
+    return machine.replacingOccurrences(of: "\0", with: "") == "arm64"
```

Comment:
I suggest use `machine.starts(with: "arm64")`, which is not only more robust and also more elegant.

Reply:
Good catch. Uploaded a new commit with this change.